gpu/test/base/gpu_executor.cu

```diff
@@ -142,6 +146,37 @@ TEST_F(GpuExecutor, CopiesDataFromGpu)
     gpu->free(orig);
 }
 
+/* Properly checks if it works only when multiple GPUs exist */
+TEST_F(GpuExecutor, PreservesStreamSettings)
+{
+    auto last_device = gko::GpuExecutor::get_num_devices() - 1;
+    cudaSetDevice(last_device);
+    auto orig = gpu->alloc<int>(2);
+    int current_stream;
+    cudaGetDevice(&current_stream);
+    ASSERT_EQ(current_stream, last_device);
+
+    gpu->free(orig);
+    cudaGetDevice(&current_stream);
+    ASSERT_EQ(current_stream, last_device);
+}
+
+TEST_F(GpuExecutor, CopiesDataFromGpuToGpu)
+{
+    int copy[2];
+    auto orig = gpu->alloc<int>(2);
+    cudaSetDevice(0);
+    init_data<<<1, 1>>>(orig);
+
+    auto copy_gpu2 = gpu2->alloc<int>(2);
+    gpu2->copy_from(gpu.get(), 2, orig, copy_gpu2);
+    cpu->copy_from(gpu2.get(), 2, copy_gpu2, copy);
+
+    EXPECT_EQ(5, copy[0]);
+    ASSERT_EQ(2, copy[1]);
+    gpu->free(copy_gpu2);
+    gpu->free(orig);
+}
 
```

Comment:
We're also missing a test which checks if the`GpuExecutor::run()` behaves properly. You can just create a dummy operation which will query if the correct device id is set when running it.

Reply:
I added this.